pkg/flowaggregator/certificate/provider.go

```diff
@@ -0,0 +1,588 @@
+// Copyright 2025 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package certificate
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/server/dynamiccertificates"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	clockutils "k8s.io/utils/clock"
+
+	"antrea.io/antrea/pkg/util/env"
+)
+
+const (
+	DefaultNamespace = "flow-aggregator"
+
+	minValidDuration = time.Hour * 24 * 90
+
+	CAConfigMapName = "flow-aggregator-ca"
+	CAConfigMapKey  = "ca.crt"
+
+	// #nosec G101: false positive triggered by variable name which includes "Secret"
+	CASecretName = "flow-aggregator-ca"
+
+	// #nosec G101: false positive triggered by variable name which includes "Secret"
+	ClientSecretName = "flow-aggregator-client-tls"
+	ServiceName      = "flow-aggregator"
+
+	SecretCertKey = "tls.crt"
+	SecretKeyKey  = "tls.key"
+)
+
+var (
+	maxAge = time.Hour * 24 * 365 // one year self-signed certs
+)
+
+type Provider interface {
+	Run(stopCh <-chan struct{})
+	GetServerCertKey() (caCertPEM []byte, serverCertPEM []byte, serverKeyPEM []byte)
+	HasSynced() bool
+	AddListener(dynamiccertificates.Listener)
```

Comment:
I don't think there is a reason to have a dependency on `dynamiccertificates` here
You can define your own Listener type